src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/PlayerDetectorPeripheral.java

```diff
@@ -212,9 +212,30 @@ public final Map<String, Object> getPlayerPos(String username) throws LuaExcepti
             return Collections.emptyMap();
 
         Map<String, Object> info = new HashMap<>();
-        info.put("x", Math.floor(existingPlayer.getX()));
-        info.put("y", Math.floor(existingPlayer.getY()));
-        info.put("z", Math.floor(existingPlayer.getZ()));
+
+        double x = existingPlayer.getX();
+        double y = existingPlayer.getY();
+        double z = existingPlayer.getZ();
+
+        if (APConfig.PERIPHERALS_CONFIG.playerSpyRandError.get()) {
+            double distance = Math.sqrt(Math.pow(x - getPos().getX(), 2) + Math.pow(y - getPos().getY(), 2) + Math.pow(z - getPos().getZ(), 2));
+
+            final int minDistance = 50;
+            final int maxDistance = 10000;
```

Comment:
Just use the `playerDetMaxRange` in config?

Reply:
I am not 100% sure what the purpose of these variables are. It seems that the error shouldn't go above 2500 if the player is further than 10000 blocks away

The code lacks a bit of documentation/comments